examples/moonshot-mainseries/01-simulate-receptors.py

```diff
@@ -0,0 +1,228 @@
+simulation = Simulation(modeller.topology, system, integrator, platform=platform)
+context = simulation.context
+context.setPositions(modeller.positions)
+
+# Write initial PDB, if requested.
+if arguments['--initial']:
+    log.info(f":page_facing_up:  Writing initial PDB to {arguments['--initial']}")
+    output_positions = context.getState(getPositions=True, enforcePeriodicBox=False).getPositions(asNumpy=True)
+    with open(arguments['--initial'], 'w') as outfile:    
+        PDBFile.writeFile(output_topology, output_positions[output_indices,:], file=outfile, keepIds=False);
+
+# Minimize energy
+log.info(':skier:  Minimizing ...')
+simulation.minimizeEnergy();
+
+# Write minimized PDB, if requested
+if arguments['--minimized']:
+    log.info(f":page_facing_up:  Writing minimized PDB to {arguments['--minimized']}")
+    output_positions = context.getState(getPositions=True, enforcePeriodicBox=False).getPositions(asNumpy=True)
+    with open(arguments['--minimized'], 'w') as outfile:    
+        PDBFile.writeFile(output_topology, output_positions[output_indices,:], file=outfile, keepIds=False);
+
+# Equilibrate
+log.info(':fire:  Heating ...')
+simulation.context.setVelocitiesToTemperature(temperature);
+simulation.step(equilibration_steps);
+
+# Add reporter to generate DCD trajectory, if requested
+if arguments['--dcdtraj']:
+    log.info(f":page_facing_up:  Will write DCD trajectory to {arguments['--dcdtraj']}")
+    from mdtraj.reporters import DCDReporter
+    simulation.reporters.append(DCDReporter(arguments['--dcdtraj'], reporting_interval, atomSubset=output_indices))
+
+# Add reporter to generate XTC trajectory, if requested
+if arguments['--xtctraj']:
+    log.info(f":page_facing_up:  Will write XTC trajectory to {arguments['--xtctraj']}")
+    from mdtraj.reporters import XTCReporter
+    simulation.reporters.append(XTCReporter(arguments['--xtctraj'], reporting_interval, atomSubset=output_indices))
+
+# Add reporter to generate PDB trajectory, if requested
+# NOTE: The PDBReporter does not currently support atom subsets
+if arguments['--pdbtraj']:
+    log.info(f":page_facing_up:  Will write PDB trajectory to {arguments['--pdbtraj']}")
+    from openmm.app import PDBReporter
+    simulation.reporters.append(PDBReporter(arguments['--pdbtraj'], reporting_interval))
+
+# Run simulation
+log.info(':coffee:  Starting simulation...')
+from rich.progress import track
+for snapshot_index in track(range(n_snapshots), ':rocket: Running production simulation...'):
+    simulation.step(reporting_interval)
+
+# Write final PDB, if requested
+if arguments['--final']:
+    log.info(f":page_facing_up:  Writing final PDB to {arguments['--final']}")
+    output_positions = context.getState(getPositions=True, enforcePeriodicBox=False).getPositions(asNumpy=True)
+    with open(arguments['--final'], 'w') as outfile:    
+        PDBFile.writeFile(output_topology, output_positions[output_indices,:], file=outfile, keepIds=False);
+
```

Comment:
would it be in scope here to implement writing out some 'representative' snapshots, i.e. do a quick clustering of the trajectory and write cluster centers to PDBs? Something like https://mdtraj.org/1.9.3/examples/centroids.html

But then again, not sure how well that scales to PL systems.

Reply:
(asking because this is an ASAP request -> would like to see quantitatively how much ligand sections like to reside in specific subpockets)

Reply:
Potentially, but I'm not certain yet whether we need to keep this particular script here.